### DISCUSSION TITLE
Flashing WiFi and IoT Hub details.

### DISCUSSION THREAD

**Discussion by djaus2** [earlephilhower/arduino-pico#2312] [Show and tell]
Have created a Sketch, as on GitHub, that can connect to WiFi in a variety of ways. Included is the ability to store details in flash (using simulated EEProm library) and read them back. Thus facilitating "headless"device model.

Also included in Azure IiT Hub Device Connection String write and read.

Connect2WiFi
============
https://github.com/djaus2/Connect2WiFi

**Comment by djaus2** [comment]
Have refactored code using namespace to simplify reuse .

**Reply by earlephilhower** [reply]
Nice work.  A couple notes you should feel very free to ignore... :laughing: 

Maybe instead of a `namespace` a `class` (and Arduino library) would be in order?  Like the `HTTPUpdate`  library/class here, for example.

Also, maybe using a `File` instead of `EEPROM` to store the config would be in order?  EEPROM really is legacy and emulated to boot (so all writes hit the exact same flash spot) and you lose everything if the Pico powers down at the wrong time.  With `LittleFS` you would get power failure protection and wear leveling.  Using `FatFS` in 4K mode you would get something you could plug into a PC and edit/copy a `config.txt` file to update.

**Comment by djaus2** [comment]
Thanks for the feedback. Will look into that implementation.

Wrt flash  wearout: I would expect though that these seeting would typically only be written once.

I do see how the FS approach would simploify updates.

**Reply by djaus2** [reply]
I have also created a separate repository [OTALedFlash](https://github.com/djaus2/OTALedFlash/) that iwa included in the [Softa-ata](https://github.com/djaus2/Soft-ata) repository. This has been discussed previously.

**Comment by djaus2** [comment]
The Connect2WiFi code has been updated to change delimeter to a a pipe and to add 2 more stored entities (5 in total). And is more easily extended to add more stored entities.
